node/http/src/api.rs

```diff
@@ -220,4 +221,20 @@ impl HttpApi {
             None => Err("block not found"),
         }
     }
+
+    pub fn get_shard_blocks_by_index(
+        &self,
+        r: &GetBlocksByIndexRequest,
+    ) -> Result<SignedShardBlocksResponse, String> {
+        let start = r.start.unwrap_or_else(|| { self.shard_chain.best_index() });
+        let limit = r.limit.unwrap_or(25);
+        match self.shard_chain.get_blocks_by_index(start, limit) {
```

Comment:
Can be rewritten as `self.shard_chain.get_blocks_by_index(start, limit).map(|blocks| ..)`